### PR TITLE
Adds cypress scaffolding and test for date input.

### PR DESCRIPTION
**Pull request by justinlittman**
## Why was this change made?
To replace Amy as date input tester.

## How was this change tested? 🤨

⚡ ⚠ If this change involves consuming from or writing to another service (or shared file system), ***run [integration test create_object_h2_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test manually in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -40,8 +40,10 @@ def ldap_groups
   def groups_from_request_env
     session['groups'] = begin
       raw_header = request.env[Settings.authorization_group_header]
-      raw_header = ENV.fetch('ROLES', nil) if Rails.env.development? # rubocop:disable Rails/EnvironmentVariableAccess
-      logger.debug("Roles are #{raw_header}")
```
Comment (lwrubel) [inline app/controllers/application_controller.rb:44]: Not sure how we use this debugging but would it be worth leaving in?

File: bin/cypress [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env ruby
+
+ENV["RAILS_ENV"] ||= "test"
+ENV["REMOTE_USER"] ||= "auser@stanford.edu"
+ENV["ROLES"] ||= "dlss:hydrus-app-administrators"
+require "pathname"
+require "cypress-rails"
+require Pathname.new(CypressRails::Config.new.dir).join("config/environment")
```
Comment (lwrubel) [inline bin/cypress:8]: Add config to run headlessly with Firefox rather than Electron, like we discussed.

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,9 +1,36 @@
...
+            - run: sudo apt-get install libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
+      - browser-tools/install-firefox
+      - node/install-packages:
+          pkg-manager: yarn
+      - run:
+          command: 'dockerize -wait tcp://localhost:5432 -timeout 1m'
+          name: Wait for DB
+      - run:
+          environment:
+            CYPRESS_RAILS_CYPRESS_OPTS: '--browser firefox'
```
Comment (lwrubel) [inline .circleci/config.yml:27]: Could remove this if it's configured to run with firefox by default when headless, as discussed.

**Comment by lwrubel** [comment]
I checked it out and it runs on my computer. 🎉 

Do you know what is causing the total-coverage check to fail?

File: README.md [hunk 4]
```diff
@@ -71,6 +71,19 @@ $ docker compose up # use -d to run in background
 
 This will spin up the H2 web application, its background workers, and all service dependencies declared in docker-compose.yml.
 
+### Cypress
+Cypress is primarily used to test features implemented with JS/Stimulus. Cypress tests are located in `cypress/spec`.
+
+To run cypress UI:
+```shell
+$ bin/cypress open
```
[the remark this answers is not in the document]
Comment (justinlittman) [reply README.md]: Done.

**Review by lwrubel — APPROVED** [review]

**Comment by justinlittman** [comment]
Good news @ndushay: It is running cypress in circle.

image

**Pull request merged**